cert/cert.go

```diff
@@ -292,3 +294,35 @@ func ParseCertAndKey(certPEM, keyPEM string) (*x509.Certificate, *rsa.PrivateKey
 	}
 	return cert, key, nil
 }
+
+// ComputePublicKey generates a public key from certPEM string
+func ComputePublicKey(certPEM string) (string, error){
+	cert, err := ParseCert(certPEM)
+	if err != nil {
+		return "", err
+	}
+	marshalledPubKey, err := x509.MarshalPKIXPublicKey(cert.PublicKey)
+	if err != nil {
+		return "", err
+	}
+	keyPEMData := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PUBLIC KEY",
+		Bytes: marshalledPubKey,
+	})
+	return string(keyPEMData), nil
+}
+
+// StoreDefaultX509Cert takes a certificate and store it in the given
+// config location, using default names.
+func StoreDefaultX509Cert(location, pem, key, public string) (error) {
+	for file, content := range map[string]string{ "/juju-cert.pem": pem, "/juju-cert.key": key,
+						      "/juju-cert.pub": public } {
+		if _, err := os.Stat(location + file); os.IsNotExist(err) {
+			err := ioutil.WriteFile(location + file, []byte(content), 0600)
```

Comment:
please use utils.AtomicWriteFile, otherwise the Stat/WriteFile pattern will be prone to half-written files